### PR TITLE
Feat(Progress): add progress component

### PR DESCRIPTION
**Pull request by Misael840**
## Summary

The progress component with the options to be circular or linear was added

## Task

- https://dd360.atlassian.net/browse/CD-1072

## Affected sections

- exports.json
- package.json
- src/components/Progress/Progress.test.tsx 
- src/components/Progress/Progress.tsx      
- src/components/Progress/ProgressCircle.tsx
- src/components/Progress/ProgressLine.tsx  
- src/components/Progress/index.ts
- src/components/Progress/progress.css
- src/components/index.ts
- src/stories/Progress.stories.tsx

## How did you test this change?

- Manually
- src/components/Progress/Progress.test.tsx 

image
image
image

### CODE REVIEW

File: src/components/Progress/Progress.test.tsx [hunk 1]
```diff
@@ -0,0 +1,209 @@
+import { it, describe, vi } from 'vitest'
+import { render } from '@testing-library/react'
+
+import Progress from './Progress'
+
+describe('<Transition/>', () => {
```
Comment (arielsrodriguez) [inline src/components/Progress/Progress.test.tsx:6]: Should I go Progress?

**Comment by arielsrodriguez** [comment]
It looks very good! 👍

**Review by arielsrodriguez — APPROVED** [review]

File: src/components/Progress/ProgressLine.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
+import React, { ReactNode, forwardRef } from 'react'
+import { composeClasses } from 'lib/classes'
+
+interface ProgressLineProps extends React.HTMLAttributes<HTMLDivElement> {
```
Comment (tbor00) [inline src/components/Progress/ProgressLine.tsx:4]: You repetead all base props on all interfaces, `ProgressCircleProps` and `ProgressProps`.
We can create the base props interface and extends it in each one

File: src/components/Progress/index.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import Progress from './Progress'
+
+export default Progress
```
Comment (tbor00) [inline src/components/Progress/index.ts:3]: well, i recommend exports components like compound components

**Pull request merged**